src/nvme/util.h

```diff
@@ -424,4 +424,149 @@ static inline void nvme_id_ns_flbas_to_lbaf_inuse(__u8 flbas, __u8 *lbaf_inuse)
+ *   have a whole-word match on the @key, and if we do, return a pointer to
+ *   the first character of value (i.e. skip leading spaces, tabs, and
+ *   equal sign)
+ *
+ * @kv: The key=value string to search for the presence of @key
+ * @key: The key to look for
+ *
+ * Return: A pointer to the first character of "value" if a match is found.
+ * NULL otherwise.
+ */
+char* kv_keymatch(const char *kv, const char *key);
+
+/**
+ * startswith - Checks that a string starts with a given prefix.
+ *
+ * @s: The string to check
+ * @prefix: A string that @s could be starting with
+ *
+ * Return: If @s starts with @prefix, then return a pointer within @s at
+ * the first character after the matched @prefix. NULL otherwise.
+ */
+char* startswith(const char *s, const char *prefix);
+
+/**
+ * round_up - Round a value @val to the next multiple speified by @mult.
+ * @val: Value to round
+ * @mult: Multiple to round to.
+ *
+ * @usage:  int x = round_up(13, sizeof(__u32)); // 13 -> 16
+ */
+#define __round_mask(val, mult) ((__typeof__(val))((mult)-1))
+#define round_up(val, mult)     ((((val)-1) | __round_mask((val), (mult)))+1)
+
+/**
+ * exat_len() - Return the size in bytes, rounded to a multiple of 4 ((i.e.
+ *         size of __u32), of the buffer needed to hold the exat value of
+ *         size @val_len.
+ */
+static inline __u16 exat_len(size_t val_len)
+{
+	return (__u16)round_up(val_len, sizeof(__u32));
+}
+
+/**
+ * exat_size - return the size of the "struct nvmf_ext_attr"
+ *   needed to hold a value of size @val_len.
+ *
+ * @val_len: This is the length of the data to be copied to the "val"
+ *         field of a "struct nvmf_ext_attr".
+ *
+ * Return: The size in bytes, rounded to a multiple of 4 (i.e. size of
+ *         __u32), of the "struct nvmf_ext_attr" required to hold a string of
+ *         length @val_len.
+ */
+static inline __u16 exat_size(size_t val_len)
+{
+	return (__u16)(SIZEOF_EXT_EXAT_WO_VAL + exat_len(val_len));
+}
+
+/**
+ * exat_ptr_next - Increment @p to the next element in the array. Extended
+ *   attributes are saved to an array of "struct nvmf_ext_attr" where each
+ *   element of the array is of variable size. In order to move to the next
+ *   element in the array one must increment the pointer to the current
+ *   element (@p) by the size of the current element.
+ *
+ * @p: Pointer to an element of an array of "struct nvmf_ext_attr".
+ *
+ * Return: Pointer to the next element in the array.
+ */
+static inline struct nvmf_ext_attr *exat_ptr_next(struct nvmf_ext_attr *p)
+{
+	return (struct nvmf_ext_attr *)
+		((uintptr_t)p + (ptrdiff_t)exat_size(le16toh(p->len)));
+}
+
+/**
+ * streq - Convenience macro to check whether two strings are equal. This
+ * is more readable than "!strcmp(a, b)".
+ *
+ * @a: first string
+ * @b: other string
+ *
+ * Example:
+ *	if (streq(string, ""))
+ *		printf("String is empty!\n");
+ */
+#define streq(a,b) (strcmp((a),(b)) == 0)
```

Comment:
First of all this define already exists in ccan, second don't use it. There is no point in papering over the fact C sucks when it comes to to strings. So everyone should know what '!strcmp()' means. streq is something I have to lookup to figure out what it does.